ps_faviconnotificationbo.php

```diff
@@ -0,0 +1,296 @@
+
+        $this->controllers = array(
+            'adminAjax' => 'AdminAjaxFaviconBO',
+        );
+
+        parent::__construct();
+
+        $this->displayName = $this->l('Order Notifications on the Favicon');
+        $this->description = $this->l('Get notified when you get new order, new client or new message directly on your browser tab of your back office even if your working on another page');
+
+        // Settings paths
+        $this->js_path = $this->_path.'views/js/';
+        $this->css_path = $this->_path.'views/css/';
+        $this->img_path = $this->_path.'views/img/';
+        $this->docs_path = $this->_path.'docs/';
+        $this->logo_path = $this->_path.'logo.png';
+        $this->module_path = $this->_path;
+        $this->ps_version = (bool)version_compare(_PS_VERSION_, '1.7', '>=');
+
+        // Confirm uninstall
+        $this->confirmUninstall = $this->l('Are you sure you want to uninstall this module?');
+        $this->ps_versions_compliancy = array('min' => '1.5', 'max' => _PS_VERSION_);
+    }
+
+    /**
+     *
+     *
+     * @param none
+     * @return bool
+     */
+    public function install()
+    {
+        Configuration::updateValue('CHECKBOX_ORDER', '1');
+        Configuration::updateValue('CHECKBOX_CUSTOMER', '1');
+        Configuration::updateValue('CHECKBOX_MESSAGE', '1');
+        Configuration::updateValue('BACKGROUND_COLOR_FAVICONBO', '#DF0067');
+        Configuration::updateValue('TEXT_COLOR_FAVICONBO', '#ffffff');
+        if (parent::install() && $this->registerHook('BackOfficeHeader') && $this->installTab() == false) {
+            return false;
+        }
+
+        return true;
+    }
+
+    public function uninstall()
+    {
+        foreach (Ps_faviconnotificationbo::$conf_fields as $field) {
+            Configuration::deleteByName($field);
+        }
+        if (parent::uninstall() &&
+            $this->uninstallTab()) {
+            return true;
+        } else {
+            $this->_errors[] = $this->l('There was an error during the uninstallation. Please contact us through Addons website');
+            return false;
+        }
+        return parent::uninstall() && $this->unregisterHook('BackOfficeHeader');
+    }
+
+    /**
+     * This method is often use to create an ajax controller
+     *
+     * @param none
+     * @return bool
+     */
+    public function installTab()
+    {
+        foreach ($this->controllers as $controller_name) {
+            $tab = new Tab();
+            $tab->active = 1;
+            $tab->class_name = $controller_name;
+            $tab->name = array();
+            foreach (Language::getLanguages(true) as $lang) {
+                $tab->name[$lang['id_lang']] = $this->name;
+            }
+            $tab->id_parent = -1;
+            $tab->module = $this->name;
+            if ($tab->add() == true) {
+                $return = true;
+            } else {
+                $return = false;
```

Comment:
With this affectation, you will have the result only of the last Tab creation.

You should init the variable `$return = true;` at the beginning of this method, then "concatenating the result of the tab creation with `$return &= $this->add();`

Reply:
There's only one admin controller to create/delete so i removed the foreach.